flink-runtime/src/main/java/org/apache/flink/runtime/execution/librarycache/BlobLibraryCacheManager.java

```diff
@@ -151,12 +163,31 @@ public URLClassLoader createClassLoader(URL[] libraryURLs) {
 				classLoaderResolveOrder,
 				libraryURLs,
 				FlinkUserCodeClassLoaders.class.getClassLoader(),
-				alwaysParentFirstPatterns);
+				alwaysParentFirstPatterns,
+				classLoadingExceptionHandler);
 		}
 	}
 
-	public static ClassLoaderFactory defaultClassLoaderFactory(FlinkUserCodeClassLoaders.ResolveOrder classLoaderResolveOrder, String[] alwaysParentFirstPatterns) {
-		return new DefaultClassLoaderFactory(classLoaderResolveOrder, alwaysParentFirstPatterns);
+	public static ClassLoaderFactory defaultClassLoaderFactory(
+			FlinkUserCodeClassLoaders.ResolveOrder classLoaderResolveOrder,
+			String[] alwaysParentFirstPatterns,
+			boolean failOnJvmMetaspaceOomError,
+			FatalErrorHandler fatalErrorHandler) {
+		return new DefaultClassLoaderFactory(
+			classLoaderResolveOrder,
+			alwaysParentFirstPatterns,
+			createClassLoadingExceptionHandler(failOnJvmMetaspaceOomError, fatalErrorHandler));
+	}
+
+	private static Consumer<Throwable> createClassLoadingExceptionHandler(
```

Comment:
If we make this method public, then we don't have to pass `failOnJvmMetaspaceOomError` and `fatalErrorHandler` to `defaultClassLoaderFactory` but could directly pass in a `Consumer<Throwable>`.

Reply:
true but the call will have to be duplicated for current use cases and I am not sure about any other at the moment. I think we can achieve the same by making the `fatalErrorHandler` nullable.